ARROW-16170: [Java][Docs] Synch java code tutorial with java cookbook (#183)
apache/arrow-cookbook: java/source/create.rst

Creating Arrow Objects
======================

A vector is the basic unit in the Arrow Java library. Data types
describe the types of values; ValueVectors are sequences of typed

> values. Vector by definition is intended to be mutable, a Vector
> can be changed it is mutable. Vector represent a one-dimensional
> sequence of homogeneous values.

Reviewer: ```suggestion
values. Vectors represent a one-dimensional sequence of values of
the same type. They are mutable containers.
```

> Vectors are provided by java arrow for the interface `FieldVector`_
> that extends `ValueVector`_.

Reviewer: ```suggestion
Vectors implement the interface `ValueVector`_. The Arrow libraries provide
implementations of vectors for various data types.
```

        listWriter.startList();
        for(int j = 0; j < 3; j++) {
            listWriter.writeInt(data[tmp_index]);
            tmp_index = tmp_index + 1;
    try(
        BufferAllocator allocator = new RootAllocator();
        ListVector listVector = ListVector.empty("listVector", allocator);
        UnionListWriter listWriter = listVector.getWriter()
    ) {
        int[] data = new int[] { 1, 2, 3, 10, 20, 30, 100, 200, 300, 1000, 2000, 3000 };
        int tmp_index = 0;
        for(int i = 0; i < 4; i++) {
            listWriter.setPosition(i);
            listWriter.startList();
            for(int j = 0; j < 3; j++) {
                listWriter.writeInt(data[tmp_index]);
                tmp_index = tmp_index + 1;
            }
            listWriter.setValueCount(3);
            listWriter.endList();
        }
        listWriter.setValueCount(3);
        listWriter.endList();
    }
    listVector.setValueCount(4);
        listVector.setValueCount(4);

    System.out.print(listVector);
        System.out.print(listVector);
    } catch (Exception e) {
        e.printStackTrace();
    }

.. testoutput::

    [[1,2,3], [10,20,30], [100,200,300], [1000,2000,3000]]

> Creating VectorSchemaRoot (Table)

Reviewer: IMO we should keep the VectorSchemaRoot example

Author: On the description mention that these objects are one-dimensional. At schema.rst there is a VectorSchemaRoot example also.
